x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/SemanticTextFieldMapper.java

```diff
@@ -124,22 +141,29 @@ public Builder setInferenceId(String id) {
             return this;
         }
 
+        public Builder setSearchInferenceId(String id) {
+            this.searchInferenceId.setValue(id);
+            return this;
+        }
+
         public Builder setModelSettings(SemanticTextField.ModelSettings value) {
             this.modelSettings.setValue(value);
             return this;
         }
 
         @Override
         protected Parameter<?>[] getParameters() {
-            return new Parameter<?>[] { inferenceId, modelSettings, meta };
+            return new Parameter<?>[] { inferenceId, searchInferenceId, modelSettings, meta };
         }
 
         @Override
         protected void merge(FieldMapper mergeWith, Conflicts conflicts, MapperMergeContext mapperMergeContext) {
-            super.merge(mergeWith, conflicts, mapperMergeContext);
+            SemanticTextFieldMapper semanticMergeWith = (SemanticTextFieldMapper) mergeWith;
+            semanticMergeWith = copySettings(semanticMergeWith, mapperMergeContext);
+
+            super.merge(semanticMergeWith, conflicts, mapperMergeContext);
```

Comment:
@jimczi This is the new approach for preserving model settings. If we detect that the new mapper doesn't have model settings or an embeddings field definition, we copy the existing model settings to the new mapper. LMKWYT!

Reply:
Yes that's much better, thanks for looking at an alternative.